cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaQuorumCheck.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.VertxUtil;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.QuorumInfo;
+
+/**
+ * Provides a method that determines whether it's safe to restart a KRaft controller, which may be the active controller
+ */
+class KafkaQuorumCheck {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaQuorumCheck.class.getName());
+    private final Reconciliation reconciliation;
+    private final Future<QuorumInfo> quorumInfoFuture;
+    private final long controllerQuorumFetchTimeoutMs;
+
+    KafkaQuorumCheck(Reconciliation reconciliation, Admin ac, Vertx vertx, long controllerQuorumFetchTimeoutMs) {
+        this.reconciliation = reconciliation;
+        this.quorumInfoFuture = describeMetadataQuorum(ac, reconciliation, vertx);
```

Comment:
Does sharing the future like this really work? I find the way of this works with creating the future in the controller and referencing it in the various methods a bit weird. I would probably expect to either wait for it here and store the info or make the call later at the point you need the information.

Reply:
Good point! I have changed this so that we call the API only when we need to.